docs/advanced-features/custom-document.md

```diff
@@ -6,7 +6,7 @@ description: Extend the default document markup added by Next.js.
 
 A custom `Document` is commonly used to augment your application's `<html>` and `<body>` tags. This is necessary because Next.js pages skip the definition of the surrounding document's markup.
 
-To override the default `Document`, create the file `./pages/_document.js` and extend the `Document` class as shown below:
+To override the default `Document`, create the file `./pages/_document.jsx` and extend the `Document` class as shown below:
```

Comment:
This doesn't appear to be a typo as `pages/_document.js` is valid and the `jsx` extension isn't required to use JSX in Next.js

Reply:
Ah okay, then apologies for calling it a typo! I didn't know you could return JSX from a non-`.jsx` file. I guess I am used to TSX files requiring the `.tsx` extension.

Reply:
Yeah makes sense, for TypeScript `.tsx` is still needed, I'm gonna close this since it does seem to be correct as is